doc/user-guide.md

```diff
@@ -53,7 +53,9 @@ $ export GO111MODULE=on
 
 ##### Vendoring
 
-The Operator SDK uses [vendoring][go_vendoring] to supply dependencies to operator projects, regardless of the dependency manager. As with the above module mode constraint, we intend to allow use of dependencies [outside of `vendor`][module_vendoring] for projects in the near future.
+The Operator SDK uses [vendoring][go_vendoring] to supply dependencies to Go operator projects if the dependency manager is `modules` and the project was initialized with the `--vendor` flag set or if the dependency manager is `dep` (which always uses vendoring).
+
+By default, the operator's dependencies are managed with `modules` and `--vendor` is not set, so calls to `go {build,clean,get,install,list,run,test}` by `operator-sdk` subcommands will use an external modules directory. Execute `go help modules` for more information.
```

Comment:
Can we move this before Line 56 so it's clear that the default behavior is `modules` with no vendoring?

And then we can state that vendoring can be used if XYZ.

Right now it's the caveats that are mentioned first.